shub_image/deploy.py

```diff
@@ -125,9 +127,12 @@ def _prepare_deploy_params(project, version, image_name, endpoint, apikey,
     if not username:
         params['pull_insecure_registry'] = True
     else:
+        # Missing password leads to auth request to registry authentication
+        # service without 'account' query parameter which breaks login
+        # procedure.
         params['pull_auth_config'] = json.dumps(
             {'username': username,
-             'password': password,
+             'password': password or ' ',
```

Comment:
It works perfectly with public images in Docker hub: username is `None` for such cases and we don't send `pull_auth_config` at all.